src/ophyd_async/plan_stubs/fly.py

```diff
@@ -6,8 +6,45 @@
 from ophyd_async.core.detector import DetectorTrigger, StandardDetector, TriggerInfo
 from ophyd_async.core.flyer import HardwareTriggeredFlyable
 from ophyd_async.core.utils import in_micros
-from ophyd_async.panda._table import SeqTable, SeqTableRow, seq_table_from_rows
-from ophyd_async.panda._trigger import SeqTableInfo
+from ophyd_async.panda import (
+    PcompDirectionOptions,
+    PcompInfo,
+    SeqTable,
+    SeqTableInfo,
+    SeqTableRow,
+    seq_table_from_rows,
+)
+
+
+def prepare_static_pcomp_flyer_and_detectors_with_same_trigger(
+    flyer: HardwareTriggeredFlyable[PcompInfo],
+    detectors: List[StandardDetector],
+    number_of_pulses: int,
+    pulse_width: int,
+    rising_edge_step: int,
+    direction: PcompDirectionOptions,
+):
+    """Prepare a hardware triggered flyable and one or more detectors.
+
+    Prepare a hardware triggered flyable and one or more detectors with the
+    same trigger. This method constructs PcompInfo from required parameters.
+    The info is required to prepare the flyer and detectors.
+
+    This prepares all supplied detectors with the same trigger.
+
+    """
+    trigger_info = PcompInfo(
+        start_postion=0,
+        pulse_width=pulse_width,
+        rising_edge_step=rising_edge_step,
+        number_of_pulses=number_of_pulses,
+        direction=direction,
+    )
```

Comment:
This is a `PcompInfo`, not a `TriggerInfo`. The `StandardDetector`s need a `TriggerInfo`, only the flyer needs the `PcompInfo`

Reply:
Thinking about this, it's not exactly clear how we generate a `TriggerInfo` from a `PcompInfo`. `number` is fine, but we don't know the `live_time` or `dead_time` as they are variable. We might have to define a `PULSE` block per detector to give it some time based pulses that it needs. For now then hardcoding these values will do, and we can tackle in another ticket.

Reply:
Actually, your solution of passing the `TriggerInfo` in from the plan neatly solves this. The plan knows how fast the motor is going. The plan knows how many trigger it asked PCOMP to make, so it can create a `TriggerInfo` telling the detector whether to use it as a trigger or a gate signal. We can defer making another ticket until we get to the double buffered sequencer position compare.